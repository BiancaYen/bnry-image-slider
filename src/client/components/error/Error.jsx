// Error

import PropTypes from 'prop-types';
import React from 'react';

// Styles
import Container from './styles';

// Prop Types
const propTypes = {
    children: PropTypes.node.isRequired
};

const Error = ({ children }) => (
    <Container>
        {children}
    </Container>
);

Error.propTypes = propTypes;

export default Error;
