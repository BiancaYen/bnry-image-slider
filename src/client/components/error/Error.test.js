// Error Test

import { mount, shallow } from 'enzyme';
import React from 'react';

// Components
import Error from './Error';

// Constants
const error = 'Error';

describe('Error', () => {
    it('renders with all required prop types', () => {
        const component = shallow(
            <Error>
                {error}
            </Error>
        );

        expect(component).toMatchSnapshot();
    });

    it('asserts that prop types are correct', () => {
        const component = mount(
            <Error>
                {error}
            </Error>
        );

        expect(component.find('div').text()).toEqual(error);
    });
});
