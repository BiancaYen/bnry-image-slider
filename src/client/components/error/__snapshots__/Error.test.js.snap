// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders with all required prop types 1`] = `
<Container>
  Error
</Container>
`;
